data_module.py

```diff
@@ -0,0 +1,123 @@
+from torchvision import transforms
+from dataset.jeopardy_dataset import JeopardyDataset
+from torch.utils.data.sampler import SubsetRandomSampler
+from torch.utils.data import DataLoader
+from pytorch_lightning import LightningDataModule
+from torch import tensor
+from PIL import ImageFilter
+import random
+import torch
+import os
+
+class GaussianBlur(object):
+    """Gaussian blur augmentation in SimCLR https://arxiv.org/abs/2002.05709"""
+    def __init__(self, sigma=(0.1, 2.0)):
+        self.sigma = sigma
+    def __call__(self, x):
+        sigma = random.uniform(self.sigma[0], self.sigma[1])
+        x = x.filter(ImageFilter.GaussianBlur(radius=sigma))
+        return x
+        
+class VQADataModule(LightningDataModule):
+  def __init__(self, batch_size, threshold=10, num_workers=8, val_split=0.2, dumb_transfer=False, num_answers=0, transfer=False, multiple_images=False):
+    super().__init__()
+    self.batch_size = batch_size
+    self.val_split = val_split
+    self.test_transform = transforms.Compose([
+        transforms.Resize(256),
+        transforms.CenterCrop(224),
+        transforms.ToTensor(),
+        transforms.Normalize(mean=[0.491, 0.482, 0.446],
+                            std=[0.247, 0.243, 0.261]),
+    ])
+    
+    if not transfer:
+        # random crop, color jitter etc 
+        self.train_transform = transforms.Compose([
+                    transforms.Resize(256),
+                    transforms.RandomResizedCrop(224, scale=(0.2, 1.)),
+                    transforms.RandomApply([
+                        transforms.ColorJitter(0.4, 0.4, 0.4, 0.1)  # not strengthened
+                    ], p=0.8),
+                    transforms.RandomGrayscale(p=0.2),
+                    transforms.RandomApply([GaussianBlur([1, 1])], p=0.5),
+                    transforms.RandomHorizontalFlip(),
+                    transforms.ToTensor(),
+                    transforms.Normalize(mean=[0.491, 0.482, 0.446],
+                            std=[0.247, 0.243, 0.261]),
+                ])
+    else:
+        # for finetuning/linear eval - confirm
+        self.train_transform = transforms.Compose([
+                    transforms.Resize(256),
+                    transforms.RandomResizedCrop(224, scale=(0.2, 1.)),
+                    transforms.RandomHorizontalFlip(),
+                    transforms.ToTensor(),
+                    transforms.Normalize(mean=[0.491, 0.482, 0.446],
+                            std=[0.247, 0.243, 0.261]),
+                ]
+                )  
+    
+    # read env variables here
+    self.questions_file = os.environ.get('QUESTIONS_FILE')
```

Comment:
Might be better to store these constants in config rather than in global variables